### PR TITLE
[PreviewCard] Created a separated single Attachment that will connect to PreviewCard

### PR DESCRIPTION
**Pull request by lewist13**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Created an attachment that will be directly connected beneath the PreviewCard.

The Attachment is currently separated from PreviewCard's view. It'll be embedded with PreviewCard as a single control at a later date to reduce redundancy and keep the code clear.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| scrolling_previewcard_5-24 | attachment_6-3_resized 26 |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/999)

### CODE REVIEW

**Comment by mischreiber** [comment]
Looks like the 40x40 image isn't actually 40x40, but 40x48. Intentional?

File: ios/FluentUI/PreviewCard/Attachment.swift [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            Image(uiImage: thumbnail)
+                .frame(width: Constants.attachmentThumbnailWidth, height: Constants.attachmentThumbnailHeight)
+                .padding(EdgeInsets(top: Constants.attachmentThumbnailPaddingTop, leading: Constants.attachmentThumbnailPaddingLeading, bottom: Constants.atachmentZeroValue, trailing: Constants.atachmentZeroValue))
+        }
+    }
+    /// Creates Attachment's entire text view.
+    @ViewBuilder
+    var attachmentText: some View {
+        VStack(alignment: .leading) {
+            Text("Text")
```
Comment (mischreiber) [inline ios/FluentUI/PreviewCard/Attachment.swift:53]: The `"Text"` and `"Subtext"` values here seem like perfect candidates to also be passed into the control at init, like the `image` value for `PreviewCard`.

File: ios/FluentUI/PreviewCard/Attachment.swift [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                .padding(EdgeInsets(top: Constants.attachmentSubtextPaddingTop, leading: Constants.atachmentZeroValue, bottom: Constants.atachmentZeroValue, trailing: Constants.atachmentZeroValue))
+        }
+        .frame(width: Constants.attachmentEntireTextWidth, height: Constants.attachmentEntireTextHeight)
+        .padding(EdgeInsets(top: Constants.atachmentZeroValue, leading: Constants.attachmentEntireTextPaddingLeading, bottom: Constants.atachmentZeroValue, trailing: Constants.atachmentZeroValue))
+    }
+    /// Creates inner Attachment view containing the thumbnail and text.
+    @ViewBuilder
+    var innerContents: some View {
+        HStack(spacing: 0) {
+        attachmentThumbnail
```
Comment (mischreiber) [inline ios/FluentUI/PreviewCard/Attachment.swift:70]: Nit: spacing looks off here.

File: ios/FluentUI/PreviewCard/Attachment.swift [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        attachmentText
+        Spacer()
+        }
+        .frame(minWidth: Constants.attachmentWidth, minHeight: Constants.attachmentHeight)
+    }
+    public var body: some View {
+        innerContents
+                /// Creates a background that matches the PreviewCard.
```
Comment (mischreiber) [inline ios/FluentUI/PreviewCard/Attachment.swift:78]: This is fine for now, but once we start to combine these controls we should probably consolidate this to only have one background, not two.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift [hunk 4]
```diff
@@ -16,18 +16,30 @@ class PreviewCardDemoController: DemoController {
     }
     override func viewDidLoad() {
         super.viewDidLoad()
-        guard let placeholderImage = UIImage.init(named: "W11_Placeholder") else {
+        guard let previewCardPlaceholder = UIImage.init(named: "PreviewCard_Placeholder") else {
+            return
+        }
+        guard let thumbnailPlaceholder = UIImage.init(named: "Thumbnail_Placeholder") else {
```
[the remark this answers is not in the document]
Comment (lewist13) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PreviewCardDemoController.swift:22]: Oh that's neat! I'll make that change.

File: ios/FluentUI/PreviewCard/Attachment.swift (hunk 3 above)
Comment (lewist13) [reply ios/FluentUI/PreviewCard/Attachment.swift:78]: No problem, I have code ready that will resolve this.

File: ios/FluentUI/PreviewCard/Attachment.swift [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    @ViewBuilder
+    var attachmentThumbnail: some View {
+        HStack {
+            Image(uiImage: thumbnail)
+                .frame(width: Constants.attachmentThumbnailWidth,
+                       height: Constants.attachmentThumbnailHeight)
+                .padding(EdgeInsets(
+                    top: Constants.attachmentThumbnailPaddingTop,
+                    leading: Constants.attachmentThumbnailPaddingLeading,
+                    bottom: Constants.atachmentZeroValue,
```
Comment (edjamesmsft) [inline ios/FluentUI/PreviewCard/Attachment.swift:52]: Should be "attachmentZeroValue" not "atachmentZeroValue"?

Comment (lewist13) [reply ios/FluentUI/PreviewCard/Attachment.swift:52]: Good eye, thanks!

File: ios/FluentUI/PreviewCard/Attachment.swift [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+@objc open class MSFAttachment: NSObject, FluentUIWindowProvider {
+    /// The UIView representing the Attachment.
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+    /// Creates a new MSFAttachment instance.
+    /// - Parameters:
+    ///   - theme: The FluentUIStyle instance representing the theme to be overriden for this Attachment.
+    ///   - thumbnail: The UIImage used as a thumbnail within the Attachment.
+    ///   - isElevated: Boolean determining if PreviewCard is flat or elevated.`
```
Comment (sophialee0416) [inline ios/FluentUI/PreviewCard/Attachment.swift:19]: Extra backquote on line 19?

Comment (sophialee0416) [inline ios/FluentUI/PreviewCard/Attachment.swift:19]: Extra backquote on line 19?

**Pull request merged**